### DISCUSSION TITLE
Need help with making large photos smaller size [SOLVED]

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#5173] [Help]
in order to upload them to a server for use on a website, i.e. b a h a i s m e x i c a l i . org

Our son takes photos with a camera and they come out as .j p g and about 3.0 Mb in size.  25 or 50 of these results in a folder contents which is too large for the web-based program to begin to display them.

i' v e been all over the net trying to find the command line to accomplish this on the approx. 25 photos in the folder, but just can't seem to locate what i' m looking for.  They just need to be large enough to be able to be displayed on a computer screen...

Thanks for your help, please stay safe and be happy!  :)

**Comment by fmw42** [comment]
Do you need them reduced in dimensions or compressed? What is your IM version and platform?  Why are 25 JPGs to large to display?  How are you displaying them?  What tool are you using?

**Reply by ghost** [reply]
First of all -  thanks for replying!

We just need them reduced from sometimes 3 Mb to 800 Kb or so - something that still looks good on a computer monitor.  It looks like Version 7.0.10.62-3 from what i can tell that Synaptic is reporting.  We are using PCLinuxOS fully updated.  On our Joomla! site we are using sigplus to display the contents of a folder in a lightbox within our articles.  We hope that this is enough info.

Please have a wonderful evening and again, thanks!  :)

**Reply by fmw42** [reply]
Use -define jpeg:extent={size}. See https://legacy.imagemagick.org/Usage/formats/#jpg_write. If you want to reduce the file size and save as a JPG.

**Reply by ghost** [reply]
Thank you very much fmw42.  That did the trick!

Have a super day and please stay safe!  :)

**Reply by michealhorn** [reply]
In Java, it is not possible to create an array of int values. The reason for this is that there are only 8 bits available in a byte. If you need to store a bigger number than 8 bits, you will have to convert your int value into an array of bytes and then back into an int again.The following example <a href="https://codeprozone.com/code/java/149835/convert-int-to-array-in-java.html">convert int to array in java</a> of bytes
java int to int array
int number = 110101; 
String temp = Integer.toString(number);
int[] numbers = new int[temp.length()];
for (int i = 0; i < temp.length(); i++) {
    numbers[i] = temp.charAt(i) - '0';
}
convert int to array in java
public class ArraySum {
    public static void main(String[] args) {
        int first = Integer.parseInt(args[0]);
        int[] firstArray = transform(first);
        int second = Integer.parseInt(args[1]);
        int[] secondArray = transform(second);
        int[] result = sum(firstArray, secondArray);
        print(result);
    }

    public static int[] transform(int num){
        // TODO change only the following part.

    }

    public static int[] sum(int[] a, int[] b){
        // TODO change only the following part.

    }

    public static void print(int[] array){
        System.out.print("[");
        for (int i = 0; i < array.length; i++){
            if(i != array.length - 1) {
                System.out.print(array[i] + ", ");
            }else{
                System.out.print(array[i] + "]");
            }
        }
    }

}

**Reply by Holly992** [reply]
Hi, you can use [VanceAI](https://vanceai.com/) to reduce photo size.
